index.js

```diff
@@ -4,14 +4,38 @@
 
 var ensureString    = require('es5-ext/object/validate-stringifiable-value')
   , resolve         = require('path').resolve
+  , deferred        = require('deferred')
   , lstat           = require('fs2/lstat')
   , renameDirectory = require('./directory')
   , renameModule    = require('./module');
 
 module.exports = function (from, to) {
 	return lstat(resolve(ensureString(from)))(function (stats) {
-		if (stats.isDirectory()) return renameDirectory(from, to);
-		if (stats.isFile()) return renameModule(from, to);
+
+		// Dealing with case sensitive move by moving the original to a temporary location
+		// then move the temporary location to the destination.
+		// @Method: Temporary rename target to .movejstmp folder
+		// then rename it again to the original target name
+		var temporaryTo;
+		var promise;
+
+		if (from !== to && from.toLowerCase() === to.toLowerCase()) {
+			temporaryTo = resolve(to, '../.movejstmp');
+			if (stats.isDirectory()) promise = deferred(renameDirectory(from, temporaryTo));
+			if (stats.isFile()) promise = deferred(renameModule(from, temporaryTo));
+		}
+
+		if (stats.isDirectory()) {
+			return deferred(promise)(function () {
+				return renameDirectory(temporaryTo || from, to);
```

Comment:
This approach unfortunately won't work (where we use different temporary name for rename), as then requires that point to original path won't be discovered (`renameDirectory` is not aware that actually should search `from` paths and not `temporaryTo` paths).

Possibly working approach would be to get between validation. So in a moment where we deal with report that `targetPath` exists. Confirm that filename of `targetPath` has same case, and if it doesn't, then just follow with operation ignoring that error.

It'll be also good to cover that in tests. So add some modules that require those paths, and test that require paths were updated after rename.